src/stream.c

```diff
@@ -955,6 +955,15 @@ int stream_update(struct stream *s)
 }
 
 
+void stream_remove_menc_media(struct stream *strm)
+{
+	if (!strm)
+		return;
+
+	strm->mes = mem_deref(strm->mes);
```

Comment:
should also set strm->menc and strm->mencs to NULL ?
otherwise it will be left in a half undefined state.

Reply:
The `strm->mencs` holds the event callback to `call.c` which gets called after the crypto parameters are replaced already. Also the stream stays secure.
The `strm->menc` can still be in place since the encryption module stays the same.

My understanding is the session stays the same while the crypto parameter changes. So the `strn->mes` changes.
Due to network timing it is possible to drop a few frames in case the receiver gets packets encrypted with the new key but has not processed the new parameters yet. It would lead to a bigger discussion about key handling and key management in the module.

Reply:
perhaps add the name "state" to the function name, to be explicit.

Please note that encryption state consist of 1 session state and N media states per media stream.
You should make sure that all states are destroyed atomically.